sdk/resource/os_unsupported.go

```diff
@@ -0,0 +1,25 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build js plan9
```

Comment:
I see your point here. The negation approach will be more future-proof, guaranteeing that the project will always build.

Including all but Linux, macOS and Windows (`!linux,!darwin,!windows`) on the `os_unsupported` file will be the right way to demonstrate on which platforms we tested on, and thus are confident the resource works.

On the other hand, I wasn't sure to preclude the availability of a working OS description for *BSD in that way. Altought I haven't tested on such platforms, _technically_ it should work.

Going with the current approach, in the wake of a new `GOOS`, it shouldn't be much of a hassle to include it either, but it can be more daunting for someone that is just learning about the project to face a compilation error at first try.

I think if we go for the "strict negation" strategy, the other `_unix` files should be stripped from all but the `linux` and `darwin` tags too.

Reply:
I don't mean to limit it to those three. I didn't want to type out everything in `_unix` in my question.

I don't have as strong opinion on what counts as "supported". Your choice to trust the sys/unix package makes sense to me.

Reply:
> I don't mean to limit it to those three. I didn't want to type out everything in _unix in my question.

Oh, got it! Got confused for a sec because Linux, macOS and Windows are the platforms with a "compatibility guarantee" (as stated in the README) so I thought you were referring exclusively to them.

Pushed this change for others to also see the final form of this approach.